### ISSUE TITLE
Create policy for finding variables that have almost the same name in the same scope.

### ISSUE THREAD

**Issue by petdance** [petdance/perl-critic-bangs#13]
```
https://twitter.com/petdance/status/697120300400578560

I'd like a static analysis tool that says "Warning: You have variables foo_bar and foobar in the same scope."
```
I found some code in the wild that was roughly:

```
$foo_stats = get_foo_stats();
$foostats = $foo_stats;
```

That's just asking for trouble, not to mention being hard to follow as a human.

My first thought is that a policy would find that `$foo_stats` and `$foostats` are the same except for the underscore.  But are there other similarities that we could detect?  What if it was `$foo_stats` and `$foo_stat` or `$foo_status`?

It should also be able to keep track of scope.  What if there's a package variable `$foo_stats` and a local variable `$foostat`?

**Comment by jonasbn** [comment]
Hello,

Have you considered using [Levenshtein distance](https://en.wikipedia.org/wiki/Levenshtein_distance)? There is a Perl implementation [available on CPAN](https://metacpan.org/pod/Text::Levenshtein).

jonasbn

**Comment by petdance** [comment]
> Have you considered using Levenshtein distance? 

That was my plan.  It's more a matter of having the time to write something.
